src/whenever/__init__.py

```diff
@@ -2590,9 +2623,27 @@ def from_common_iso8601(cls, s: str, /) -> UTCDateTime:
         Use :meth:`OffsetDateTime.from_common_iso8601` if you'd like to
         parse an ISO 8601 string with a nonzero offset.
         """
-        if s[10] != "T" or s.endswith(("z", "-00:00")):
-            raise InvalidFormat()
-        return cls._from_py_unchecked(_parse_utc_rfc3339(s))
+        try:
+            if s[10] == "T" and not s.endswith(("z", "-00:00")):
+                return cls._from_py_unchecked(_parse_utc_rfc3339(s))
+            else:
+                # Examine the string again to keep the above happy path fast
```

Comment:
Let's not put effort into highly customized messages here. Let's stick to one generic message.

Reasoning: 
- to be consistent we'd have to do this everywhere
- this would take a big effort
- the standard library also just has one generic exception
- customized messages have little use—the docstring can easily be consulted on why the string is invalid.

Reply:
Makes sense. Originally I didn't set off to add more specific error messages, but arrived at it by iterating on ideas to keep code duplication to a minimum while raising the same error from different points of potential failure in a given function. With the introduction of utility functions to generate errors this is no longer necessary.